modules/vnet/outputs.tf

```
@@ -5,3 +5,11 @@ output "location" {
 output "resource_group_name" {
   value = azurerm_resource_group.vnet.name
 }
+
+output "subnet_ids" {
+  value = module.vnet.vnet_subnets
+}
+
+output "network_security_group_name" {
+  value = azurerm_network_security_group.vnet_southcentralus.name
```

This should fix the current plan issues

```suggestion
  value = azurerm_network_security_group.nsg.name
```